packages/components/label/src/label.tsx

```diff
@@ -6,21 +8,69 @@ import { FormattedMessage } from 'react-intl';
 import requiredIf from 'react-required-if';
 import RequiredIndicator from './required-indicator';
 
-const Label = (props) => {
+type TBasicTextProps = {
+  intlMessage?: MessageDescriptor;
+  children?: React.ReactNode;
+};
+type TLabelProps = {
+  /**
+   * Use this to set the relationships between the label and a non-form element.
+   * For example:
+   *   <label id="foo">
+   *   <div aria-labelledby="foo">
+   */
+  id?: string;
+  /**
+   * Use this to set the relationships between the label and a form element.
+   * For example:
+   *   <label for="foo">
+   *   <input id="foo">
+   *
+   *   <label id="foo">
+   *   <input aria-labelledby="foo">
+   */
+  htmlFor?: string;
+  isBold?: boolean;
+  isRequiredIndicatorVisible?: boolean;
+  tone?: 'primary' | 'inverted';
+} & TBasicTextProps;
+
+const Label = (props: TLabelProps) => {
   if (props.id && props.htmlFor) {
     invariant(
       false,
       `ui-kit/Label: provide only the "id" or the "htmlFor" properties, not both.`
     );
   }
+
+  const hasContent =
+    props.intlMessage || Boolean(React.Children.count(props.children));
+
+  invariant(
+    hasContent,
+    [
+      'Warning: Failed prop type:',
+      `The prop \`intlMessage\` is marked as required in \`Label\``,
+      'but its value is `undefined`',
+    ].join(' ')
+  );
+  invariant(
+    hasContent,
+    [
+      'Warning: Failed prop type:',
+      `The prop \`children\` is marked as required in \`Label\``,
+      'but its value is `undefined`',
+    ].join(' ')
+  );
```

Comment:
This is the same as [`warnIfMissingContent`](https://github.com/commercetools/ui-kit/blob/main/packages/components/text/src/text.tsx#L40:L63) but I don't see a strong reason yet to extract it into a util.

Reply:
Don't do it until it hurts :)

Reply:
Can't we combine the message into one?

Reply:
I keep them separate. I find them easier to structure

Reply:
The point is that the user needs to provide one or the other. If you show two separate warning messages, it seems like both props are required, which is not true.

Reply:
@adnasa ⬆️?

Reply:
My bad. missed this comment.
yeah you made a good point. will address it

Reply:
9ae1868